pkg/controller/daemon/daemon_controller.go

```diff
@@ -721,13 +721,6 @@ func (dsc *DaemonSetsController) updateNode(old, cur interface{}) {
 	oldNode := old.(*v1.Node)
 	curNode := cur.(*v1.Node)
 
-	if reflect.DeepEqual(oldNode.Labels, curNode.Labels) &&
```

Comment:
Nodes heartbeat every 10 seconds, updating the pod status. This change makes the ds controller evaluate `nodeShouldRunDaemonPod` at a frequency of `<num nodes> * <num daemonsets> / 10 seconds`. I agree that this logic is prone to breaking as written, but I'm worried about the cost we are adding. Instead of maintaining a whiltelist of node fields that should re-trigger a sync or removing the fastpath altogether, can we implement a blacklist? e.g. zero out the noisy fields we don't care about like the timestamp on the heartbeat, then do a reflect.DeepEqual on the entire object.

Reply:
Agree with this approach.

Reply:
Done. PTAL!